libmcount/mcount.c

```diff
@@ -787,15 +787,21 @@ static void mcount_finish(void)
 
 static bool mcount_check_rstack(struct mcount_thread_data *mtdp)
 {
-	if (mtdp->idx >= mcount_rstack_max) {
-		static bool warned = false;
+	static bool warned = false;
 
+	if (mtdp->idx >= mcount_rstack_max) {
 		if (!warned) {
+			struct mcount_ret_stack *rstack;
+
 			pr_warn("too deeply nested calls: %d\n", mtdp->idx);
+			/* flush current rstack */
+			rstack = &mtdp->rstack[mtdp->idx - 1];
```

Comment:
What if idx was increased more than the `mcount_rstack_max`?

Reply:
I thought that `mtdp->idx` is only increased one by one, so never be beyond `mcount_rstack_max` in line 798 because of the if statement and `warned` variable. I just changed it to `mcount_rstack_max - 1` anyway.